### PR TITLE
abuse/signup: Add CAPTCHA API & security analysis

### PR DESCRIPTION
**Pull request by KellerFuchs**
I need some people to double-check the API design and security claims.

### CODE REVIEW

**Comment by KellerFuchs** [comment]
First issue: this makes the assumption that the CAPTCHAs are “hard” (in the sense that they cannot be likely solved within a few tries).
For TextCaptcha, I'm unconvinced it is true.

We can weaken the assumption to “cannot be solved on the first try”, but this requires keeping a bit of (in-memory) state, which might make it a bit more annoying to scale `api.hashbang.sh` in the future.
One possible solution would be to bind the token to the API instance, and require that clients contact the same instance to actually perform the registration.

**Comment by KellerFuchs** [comment]
@hashbang/administrators @hashbang/contributers Penny for your thoughts?

File: abuse/signup.md [hunk 1]
```diff
@@ -46,8 +46,81 @@ channels are in English.
...
+  randomly generated when the API server starts (and never persisted to
+  disk);
+- the encrypted data is Base64-encoded, using the
+  [RFC 4648 URL-safe alphabet](https://tools.ietf.org/html/rfc4648#section-5).
+
+The use of a random, volatile key for token encryption implies two trade-offs:
+
+- CAPTCHA tokens are implicitely expired when the application is restarted;
+  they do expire within a short timeframe anyway, mitigating the issue;
+- `api.hashbang.sh` can only be served by a single instance.
```
Comment (lrvick) [inline abuse/signup.md:116]: There is nothing with this design that prevents > 1 instances. We would simply employ sticky-sessions on the load balancer.

**Comment by lrvick** [comment]
On paper at least, this looks solid. +1

**Pull request merged**